### PR TITLE
Submit link references with release notes

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
Hello! Thanks for `chandler`, it's very useful!

We'll be using it to submit release notes to github. It works great except for a slight issue. We use [link reference definitions](https://github.github.com/gfm/#link-reference-definitions) for pull requests and users in our changelog in order to keep the raw version of it more readable. However, `chandler` does not submit them to GitHub, so github is unable to interpret them and show the proper links. The links still work because of the autolinking github feature, but they are displayed with ugly brackets around them. I think the issue would be fixed if the link reference definitions were submitted to github in addition to the content of each release, and that's what this PR implements.

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
Hei @mattbrictson, any thoughts about this? I've used it for a couple of releases so far and it works great! :)

File: lib/chandler/changelog.rb [hunk 1]
```diff
@@ -72,11 +72,18 @@ def versions_at_headings(heading_re)
...
-        versions[version.version_number] = text if version
+        text_with_references = [text, link_references].join("\n\n")
+        versions[version.version_number] = text_with_references if version
       end
     end
     # rubocop:enable Style/SymbolProc
 
+    def link_references
+      @link_references ||= text.split("\n")
+                               .select { |line| line =~ /^\[.*\]: .*/ }
```
Comment (deivid-rodriguez) [inline lib/chandler/changelog.rb]: Maybe I should add "http" to the regexp to make it a little bit stronger?

Comment (deivid-rodriguez) [reply lib/chandler/changelog.rb]: Fixed in 0a9487245bb384b1d051a7f929c480e5bbd2a431.

**Comment by deivid-rodriguez** [comment]
Rebased to play nice with recent changes in master.

File: test/fixtures/changelog/activeadmin.md [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+[@stefsava]: https://github.com/stefsava
+[@tiagotex]: https://github.com/tiagotex
+[@timoschilling]: https://github.com/timoschilling
+[@TimPetricola]: https://github.com/TimPetricola
+[@varyonic]: https://github.com/varyonic
+[@wasifhossain]: https://github.com/wasifhossain
+[@Wowu]: https://github.com/Wowu
+[@wspurgin]: https://github.com/wspurgin
+[@zorab47]: https://github.com/zorab47
+[@chrp]: https://github.com/chrp
```
Comment (deivid-rodriguez) [inline test/fixtures/changelog/activeadmin.md]: This is part of a real changelog, that's why it's quite big. But if you want a more minimal file, I can reduce it.

**Comment by deivid-rodriguez** [comment]
@mattbrictson Any thoughts about this? Even something like "I like the idea but I need to find some time to review it" or like "No way I'll add this to chandler" will help! Thanks!

**Comment by mattbrictson** [comment]
Actually, this is great 👍

So sorry I didn't review this when it first came around. I haven't worked on chandler it quite a while so I'm not prepared to do a deep dive into the implementation. However I see that your change is concise and has good test coverage, so this seems like an easy one to approve. Thanks for the contribution!

I will likely release a new version of Chandler this weekend that includes this.

**Comment by mattbrictson** [comment]
Note that the next chandler release will require Ruby 2.3+. Hope that's OK 😬

**Comment by deivid-rodriguez** [comment]
It is! Thanks so much @mattbrictson!

**Comment by mattbrictson** [comment]
🚀 0.9.0 released https://github.com/mattbrictson/chandler/releases/tag/v0.9.0

**Pull request merged**